examples/xgboost_simple.py

```diff
@@ -28,16 +30,35 @@ def objective(trial):
     param = {
         "verbosity": 0,
         "objective": "binary:logistic",
+        #use exact for small dataset
+        "tree_method": "exact",
+        #evaluation metric
+        "eval_metric": trial.suggest_categorical("eval_metric", ["rmse", "rmsle", "logloss","map"]),
+        #defines booster, gblinear for linear functions 
         "booster": trial.suggest_categorical("booster", ["gbtree", "gblinear", "dart"]),
+         #L2 regularization weight
         "lambda": trial.suggest_float("lambda", 1e-8, 1.0, log=True),
+        #L1 regularization weight
         "alpha": trial.suggest_float("alpha", 1e-8, 1.0, log=True),
+        #sampling ratio for training data
+        'subsample': trial.suggest_float('subsample', 0.1,1.0),
+        #sampling according to each tree
+        'colsample_bytree': trial.suggest_float('colsample_bytree', 0.1,1.0),
     }
 
     if param["booster"] == "gbtree" or param["booster"] == "dart":
-        param["max_depth"] = trial.suggest_int("max_depth", 1, 9)
+        #maximum depth of the tree,signifies complexity of the tree
+        param["max_depth"] =  trial.suggest_int("max_depth", 3, 9)
```

Comment:
Why do you change the lower bound of `max_depth`?

Reply:
Oops, sorry it might come from [this page](https://www.analyticsvidhya.com/blog/2016/03/complete-guide-parameter-tuning-xgboost-with-codes-python/)